src/runtime/virtcontainers/qemu.go

```diff
@@ -1001,6 +1032,65 @@ func (q *qemu) LogAndWait(qemuCmd *exec.Cmd, reader io.ReadCloser) {
 	qemuCmd.Wait()
 }
 
+func (q *qemu) AttestVM(ctx context.Context) error {
+	if q.arch.guestProtection() != sevProtection {
+		return nil
+	}
+
+	if err := q.qmpSetup(); err != nil {
+		return err
+	}
+	kernelPath, err := q.config.KernelAssetPath()
+	if err != nil {
+		return err
+	}
+
+	initrdPath, err := q.config.InitrdAssetPath()
+	if err != nil {
+		return err
+	}
+
+	firmwarePath, err := q.config.FirmwareAssetPath()
+	if err != nil {
+		return err
+	}
+
+	kernelParameters := q.kernelParameters()
+	launchId := q.arch.getSEVPreAttestationId()
+
+	// Guest must be paused so that secrets can be injected.
+	// Guest will be continued by the Attestation function
+	sevConfig := kbs.GuestPreAttestationConfig{
+		Proxy:            q.config.GuestPreAttestationURI,
+		Policy:           q.config.SEVGuestPolicy,
+		Keyset:           q.config.GuestPreAttestationKeyset,
+		LaunchId:         launchId,
+		KernelPath:       kernelPath,
+		InitrdPath:       initrdPath,
+		FwPath:           firmwarePath,
+		KernelParameters: kernelParameters,
+	}
+
+	if q.config.GuestPreAttestationMode == kbs.Online {
+		sevConfig.SecretGuid = kbs.OnlineSecretGuid
+		sevConfig.SecretType = kbs.OnlineSecretType
+	} else if q.config.GuestPreAttestationMode == kbs.Offline {
```

Comment:
I think we would basically just need to remote this conditional if we want to drop offline support.